### ISSUE TITLE
Biotite is unable to read alignments from Fasta files

### ISSUE THREAD

**Issue by danijoo** [biotite-dev/biotite#26]
Popular sequence alignment programs like muscle use the `_` (underscore) symbol to represent gaps in the alignment. However, Biotite only recognices the `*` symbol. For convience, fasta io should work with both formats and silently convert the different gap formats.

**Comment by padix-key** [comment]
In a Biotite `ProteinSequence` the `*` symbol depicts the stop signal, not a gap. A gap means the absence of any symbol at that position, hence an alignment cannot be directly represented by a `Sequence` object.
Instead `sequence.align.Alignment` objects are used for this purpose. These objects contain the ungapped `Sequence` objects and a trace, that contains indices to these `Sequence` objects, describing which symbols are aligned. Please refer to the Biotite documentation for more information about `Alignment` objects.

Unfortunately there is currently no one-line-of-code way to load an `Alignment` from a `FastaFile`, but this is definitely something that should be added. I will prepare a PR for this. For now, the following short script should do the job:

```
import biotite.sequence as seq
import biotite.sequence.align as align
import biotite.sequence.io.fasta as fasta

# An example Fasta file containing an alignemnt
fasta_file = fasta.FastaFile()
fasta_file["seq1"] = "ADTRCGTARDCGTR-DRTCGRAGD"
fasta_file["seq2"] = "ADTRCGT---CGTRADRTCGRAGD"
fasta_file["seq3"] = "ADTRCGTARDCGTRADR--GRAGD"

# The actual script
# Create an alignment object from the Fasta file
# Obtain the gapped sequences
seq_strings = [seq_str for name, seq_str in fasta_file]
# The sequences are the gapped sequences in the alignemt
# without the gaps
sequences = [seq.ProteinSequence(seq_str.replace("-",""))
                for seq_str in seq_strings]
# The trace can be created using the static 'trace_from_strings()'
trace = align.Alignment.trace_from_strings(seq_strings)
# Finally create the alignment
alignment = align.Alignment(sequences, trace, score=None)
print(alignment)
```

Output:

```
ADTRCGTARDCGTR-DRTCGRAGD
ADTRCGT---CGTRADRTCGRAGD
ADTRCGTARDCGTRADR--GRAGD
```

To my knowledge MUSCLE uses `-` symbols for gaps. If in your case `_` is used, you have write `seq_str.replace("_","")`.

**Comment by padix-key** [comment]
Fixed via #28

**Comment by padix-key** [comment]
Functionality is affected by #29.

**Comment by padix-key** [comment]
#29 is fixed via #30.

**Comment by padix-key** [comment]
You might also have a look at the functions introduced via PR #31, especially `get_codes()`.
